### PR TITLE
F 536 implement subscribe feature

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
@aminlatifi 

Plz check netlify to is it good or not ( its not ideal and has basic UI)

### CODE REVIEW

File: src/components/ProjectSubscription.jsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+  const { authenticated } = currentUser;
+
+  const toggleSubscription = async enabled => {
+    if (!authenticated) {
+      React.signIn();
+      return;
+    }
+    setLoading(true);
+    await actionWithLoggedIn(currentUser);
```
Comment (aminlatifi) [inline src/components/ProjectSubscription.jsx:24]: It's not necessary, upper condition on authenticated is doing the same job

File: src/components/ProjectSubscription.jsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        enabled,
+      });
+      setSubscribed(result.enabled);
+      React.toast.success(
+        enabled
+          ? 'You have been subscribed successfully'
+          : 'You have been unsubscribed successfully',
+      );
+    } catch (err) {
+      ErrorHandler(err, 'Something wrong in updating user roles, please try later');
```
Comment (aminlatifi) [inline src/components/ProjectSubscription.jsx:38]: Update text

**Comment by aminlatifi** [comment]
https://fontawesome.bootstrapcheatsheets.com/

**Comment by aminlatifi** [comment]
https://www.w3schools.com/howto/howto_css_icon_buttons.asp

**Comment by aminlatifi** [comment]
https://getbootstrap.com/docs/4.0/components/buttons/#outline-buttons

File: src/components/ProjectSubscription.jsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  projectTypeId: PropTypes.string,
+  projectType: PropTypes.oneOf(['campaign', 'milestone', 'dac']),
+};
+
+ProjectSubscription.defaultProps = {
+  projectTypeId: '',
+  projectType: '',
+};
+
+const isEqual = (prevProps, nextProps) => {
```
Comment (aminlatifi) [inline src/components/ProjectSubscription.jsx:104]: You can clear this one

**Review by aminlatifi — CHANGES_REQUESTED** [review]

File: src/components/ProjectSubscription.jsx (hunk 1 above)
Comment (mohammadranjbarz) [reply src/components/ProjectSubscription.jsx]: done

File: src/components/ProjectSubscription.jsx (hunk 2 above)
Comment (mohammadranjbarz) [reply src/components/ProjectSubscription.jsx]: done

File: src/components/ProjectSubscription.jsx (hunk 3 above)
Comment (mohammadranjbarz) [reply src/components/ProjectSubscription.jsx]: done

**Comment by mohammadranjbarz** [comment]
@aminlatifi  It's the new styles and ready for merge, you can check it here:

https://deploy-preview-1922--giveth-dapp.netlify.app

Screen Shot 1399-12-26 at 18 42 16

Screen Shot 1399-12-26 at 18 42 26

**Comment by aminlatifi** [comment]
If I am not authenticated it doesn't do anything

**Comment by mohammadranjbarz** [comment]
> If I am not authenticated it doesn't do anything
> @MohammadPCh Please do something like this:
> 
> ```
>     const authenticated = await authenticateUser(currentUser, false);
> 
>     if (authenticated) {
>     /// Do the logic
>     }
> ```
@aminlatifi  done

**Pull request merged**